### PR TITLE
Add database request

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -430,6 +433,29 @@ public void onMessage(Message message) {
 	}
 }
 ```
+<a name="databaseRequest" />
+
+### Trace SQL database requests
+
+A SQL database request is traced by calling `traceSqlDatabaseRequest`. For details about usage see the [OneAgentSDK specification](https://github.com/Dynatrace/OneAgent-SDK#database)
+
+```
```
Comment (arminru) [inline README.md]: for syntax highlighting:
```suggestion
```java
```

File: samples/README.md [hunk 2]
```diff
@@ -6,8 +6,9 @@ Sample applications showing how to use Dynatrace OneAgent SDK for Java to create
 
 - `remotecall`: shows usage of remote call API. Allows you to tag remote calls in the same or between different JVMs.  
 - `in-process-linking`: shows usage of in-process-linking API. Allows you to tag eg. asynchronous execution inside the same process.
-- `webrequest`: shows usage of outging- and incoming webrequests.
+- `webrequest`: shows usage of outging- and incoming webrequests
```
Comment (arminru) [inline samples/README.md]:
```suggestion
- `webrequest`: shows usage of outgoing and incoming web request tracing
```

File: src/main/java/com/dynatrace/oneagent/sdk/api/DatabaseRequestTracer.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.oneagent.sdk.api;
+
+/**
+ * Interface for outgoing database tracer.
+ * <a href="https://github.com/Dynatrace/OneAgent-SDK#database">https://github.com/Dynatrace/OneAgent-SDK#database</a>
```
Comment (arminru) [inline src/main/java/com/dynatrace/oneagent/sdk/api/DatabaseRequestTracer.java]:
```suggestion
 * <a href="https://github.com/Dynatrace/OneAgent-SDK-for-Java#databaseRequest">https://github.com/Dynatrace/OneAgent-SDK-for-Java#databaseRequest</a>
```

File: README.md [hunk 4]
```diff
@@ -430,6 +433,29 @@ public void onMessage(Message message) {
...
 }
 ```
+<a name="databaseRequest" />
+
+### Trace SQL database requests
+
+A SQL database request is traced by calling `traceSqlDatabaseRequest`. For details about usage see the [OneAgentSDK specification](https://github.com/Dynatrace/OneAgent-SDK#database)
+
+```
+String sql = "SELECT * FROM transformationdata WHERE transformation_id = " + id;
```
Comment (arminru) [inline README.md:443]:
```suggestion
String sql = "SELECT * FROM transformationdata WHERE transformation_id = " + id; // for brevity, don't do that ;)
```

File: src/main/java/com/dynatrace/oneagent/sdk/api/enums/MessageSystemVendor.java [hunk 5]
```diff
@@ -19,7 +19,9 @@
 
 /**
  * Enumerates all well-known messaging systems. See {@link OneAgentSDK#createMessagingSystemInfo(String, String, MessageDestinationType, ChannelType, String)}.
- * Using these constants ensures that services captured by OneAgentSDK are handled the same way as traced via built-in sensors. 
+ * Using these constants ensures that services captured by OneAgentSDK are handled the same way as traced via built-in sensors.
```
Comment (arminru) [inline src/main/java/com/dynatrace/oneagent/sdk/api/enums/MessageSystemVendor.java:22]:
```suggestion
 * Using these constants ensures that services captured by OneAgentSDK are handled the same way as if they were traced by the OneAgent out-of-the-box.
```

**Review by arminru — APPROVED** [review]
see comments, otherwise LGTM

**Review by SonjaChevre — APPROVED** [review]
LGTM

File: src/main/java/com/dynatrace/oneagent/sdk/impl/OneAgentSDKFactoryImpl.java [hunk 6]
```diff
@@ -30,13 +30,10 @@
...
 	static final int oneSdkMajor = 1;
-	static final int oneSdkMinor = 5;
+	static final int oneSdkMinor = 7;
 	static final int oneSdkFix = 0;
 
-	/** the only SDK instance (one for every classLoader) */
-	private static OneAgentSDK oneSDK = null;
-
-	public static boolean debugOneAgentSdkStub = false;
+	public static final boolean debugOneAgentSdkStub = Boolean.parseBoolean(System.getProperty("com.dynatrace.oneagent.sdk.debug", "false"));
```
Comment (Oberon00) [inline src/main/java/com/dynatrace/oneagent/sdk/impl/OneAgentSDKFactoryImpl.java:36]: What if someone sets this system property to an invalid boolean value, e.g. "on"? Won't this result in an user facing exception?

File: src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/DatabaseInfoImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	private final String channelEndpoint;
+
+	DatabaseInfoImpl(String name, String vendor, ChannelType channelType, String channelEndpoint) {
+		this.name = name;
+		this.vendor = vendor;
+		this.channelType = channelType;
+		this.channelEndpoint = channelEndpoint;
+	}
+	
+	String getName() {
```
Comment (Oberon00) [inline src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/DatabaseInfoImpl.java:35]: Why no explicit `private` here?

File: src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/OneAgentSDKProxy.java [hunk 8]
```diff
@@ -268,4 +272,32 @@ public IncomingMessageProcessTracer traceIncomingMessageProcess(MessagingSystemI
...
+				OneAgentSDKFactoryImpl.logDebug("- invalid DatabaseInfo object provided: "
+						+ (databaseInfo == null ? "null" : databaseInfo.getClass().getName()));
+			}
+			return DatabaseRequestTracerNoop.INSTANCE;
+		}
+
+		Object agentObject = apiProxy.oneAgentSDK_traceSQLDatabaseRequest(agentSdkImpl, (DatabaseInfoImpl) databaseInfo, statement);
+		if (agentObject == null) {
+			if (OneAgentSDKFactoryImpl.debugOneAgentSdkStub) {
+				OneAgentSDKFactoryImpl.logDebug("- OneAgent failed to provide provide object");
```
Comment (Oberon00) [inline src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/OneAgentSDKProxy.java:296]: typo "provide provide". Also, maybe "failed to create database request trace object" would be more informative. And why the leading "-"?

File: README.md [hunk 9]
```diff
@@ -430,6 +433,29 @@ public void onMessage(Message message) {
...
+
+DatabaseInfo databaseInfo = oneAgentSdk.createDatabaseInfo("TransformationDb", DatabaseVendor.FIREBIRD.getVendorName(), ChannelType.TCP_IP, "db-serv01.acme.com:2323");
+
+DatabaseRequestTracer databaseTracer = oneAgentSdk.traceSqlDatabaseRequest(databaseInfo, sql);
+databaseTracer.start();
+try {
+	Result result = executeTheDatabaseCall(sql);
+	databaseTracer.setRowCountReturned(result.getRows().getLength());
```
Comment (Oberon00) [inline README.md]: Wrong name: must be `setReturnedRowCount`.

**Review by Oberon00 — CHANGES_REQUESTED** [review]

File: src/main/java/com/dynatrace/oneagent/sdk/impl/OneAgentSDKFactoryImpl.java (hunk 6 above)
Comment (ghost) [reply src/main/java/com/dynatrace/oneagent/sdk/impl/OneAgentSDKFactoryImpl.java:36]: default is false :)

File: src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/OneAgentSDKProxy.java (hunk 8 above)
Comment (ghost) [reply src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/OneAgentSDKProxy.java:296]: thank - will be included in next release

File: README.md (hunk 9 above)
Comment (ghost) [reply README.md]: done

**Review by Oberon00 — APPROVED** [review]
LGTM

**Pull request merged**